spring-integration-core/src/main/java/org/springframework/integration/channel/NullChannel.java

```diff
@@ -27,32 +32,144 @@
  * All receive() calls will return <em>null</em>, and all send() calls
  * will return <em>true</em> although no action is performed.
  * Note however that the invocations are logged at debug-level.
- * 
+ *
  * @author Mark Fisher
+ * @author Gary Russell
  */
-public class NullChannel implements PollableChannel {
+@ManagedResource
+public class NullChannel implements PollableChannel, MessageChannelMetrics, NamedComponent {
 
 	private final Log logger = LogFactory.getLog(this.getClass());
 
+	private final ChannelSendMetrics metrics = new ChannelSendMetrics("nullChannel");
+
+	private volatile boolean statsEnabled;
+
+	@Override
+	public String getComponentName() {
+		return "nullChannel";
+	}
+
+	@Override
+	public String getComponentType() {
+		return "channel";
+	}
+
+	@Override
+	public void reset() {
+		metrics.reset();
+	}
+
+	@Override
+	public void enableStats(boolean statsEnabled) {
+		this.statsEnabled = statsEnabled;
+	}
+
+	@Override
+	public boolean isStatsEnabled() {
+		return this.statsEnabled;
+	}
+
+	@Override
+	public int getSendCount() {
+		return metrics.getSendCount();
+	}
+
+	@Override
+	public long getSendCountLong() {
+		return metrics.getSendCountLong();
+	}
+
+	@Override
+	public int getSendErrorCount() {
+		return metrics.getSendErrorCount();
```

Comment:
All these `error` metrcis don't have value for `NullChannel`, although I think it won't hurt just delegate as you do it now.